### PR TITLE
Window title update

### PR DESCRIPTION
**Pull request by medek**
Added window title updating for redraw update "set_title" in
handleRedraw.

Also changed the cmake so binaries end up in buid/bin and libraries in
build/lib.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -13,6 +13,9 @@ endif()
 set(CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake")
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 
+set(LIBRARY_OUTPUT_PATH ${PROJECT_BINARY_DIR}/lib)
+set(EXECUTABLE_OUTPUT_PATH ${PROJECT_BINARY_DIR}/bin)
+
```
Comment (equalsraf) [inline CMakeLists.txt]: Just for clearness, can you split this into a separate commit?

File: src/gui/shell.h [hunk 2]
```diff
@@ -20,7 +20,8 @@ class Shell: public QWidget
 	QSize sizeHint() const;
 	static QColor color(qint64 color, const QColor& fallback=QColor());
 	virtual QVariant inputMethodQuery(Qt::InputMethodQuery) const;
-
+signals:
+	void neovimSetTitle(const QString &title);
```
Comment (equalsraf) [inline src/gui/shell.h]: Can we rename this to `neovimTitleChanged()`?

File: src/gui/shell.cpp [hunk 3]
```diff
@@ -421,6 +423,11 @@ void Shell::handleInsertMode(QPainter& painter)
 	m_insertMode = true;
 }
 
+void Shell::handleSetTitle(const QVariantList& opargs)
+{
+	emit neovimSetTitle(opargs.at(0).toString());
+}
```
Comment (equalsraf) [inline src/gui/shell.cpp]: Be a bit more pedantic (`QList::at()` needs to be passed a valid index) and take care of the string encoding conversions

```diff
diff --git a/src/gui/shell.cpp b/src/gui/shell.cpp
index 47a34be..aa0d844 100644
--- a/src/gui/shell.cpp
+++ b/src/gui/shell.cpp
@@ -425,7 +425,13 @@ void Shell::handleInsertMode(QPainter& painter)
 
 void Shell::handleSetTitle(const QVariantList& opargs)
 {
-       emit neovimSetTitle(opargs.at(0).toString());
+
+       if (opargs.size() != 1 || !opargs.at(0).canConvert<QByteArray>()) {
+               qWarning() << "Unexpected arguments for set_title:" << opargs;
+               return;
+       }
+       QString title = m_nvim->decode(opargs.at(0).toByteArray());
+       emit neovimSetTitle(title);
 }
```

**Comment by equalsraf** [comment]
Cheers @medek thanks for the patches, added a few comments.

**Comment by medek** [comment]
Split the commit. Added the additional code to handleSetTitle().
I actually didn't know about m_nvim->decode, whoops :P.

**Comment by equalsraf** [comment]
Merged, thanks.
